modules/search/class.jetpack-search.php

```diff
@@ -692,11 +689,61 @@ function convert_wp_es_to_es_args( array $args ) {
 			 *     'Post Type' => array( 'type' => 'post_type', 'count' => 10 ) ),
 			 * );
 			 */
-			'aggregations'         => null,
+			'aggregations'   => null,
 		);
 
 		$args = wp_parse_args( $args, $defaults );
 
+		$parser = new Jetpack_WPES_Search_Query_Parser( $args['query'], array( get_locale() ) );
+
+		$match_content_fields = $parser->merge_ml_fields(
+			array(
+				'title'    => 2,
+				'content'  => 2,
+				'author'   => 1,
+				'tag'      => 1,
+				'category' => 1,
+			),
+			array()
+		);
+
+		$boost_content_fields = $parser->merge_ml_fields(
+			array(
+				'title'       => 2,
+				'description' => 1,
+				'tags'        => 1,
+			),
+			array(
+				'author_login^2',
+				'author^2',
+			)
+		);
+
+		$parser->phrase_filter( array(
+			'must_query_fields'  => $match_content_fields,
+			'boost_query_fields' => $boost_content_fields,
+		) );
+		$parser->remaining_query( array(
+			'must_query_fields'  => $match_content_fields,
+			'boost_query_fields' => $boost_content_fields,
+		) );
+
+		// Boost on phrases
+		$parser->remaining_query( array(
+			'boost_query_fields' => $boost_content_fields,
+			'boost_query_type'   => 'phrase',
+		) );
+
+		// Other boosts
+		$parser->add_max_boost_to_functions( 20 );
+		$parser->add_function( 'field_value_factor', array(
```

Comment:
This field is the same across all posts, so let's leave off any decay except for on based on date_gmt